addons/hr_recruitment_survey/models/survey_survey.py

```diff
@@ -0,0 +1,23 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class SurveySurvey(models.Model):
+    _inherit = 'survey.survey'
+
+    survey_type = fields.Selection(selection_add=[('recruitment', 'Recruitment')], ondelete={'recruitment': 'set default'})
+    hr_job_ids = fields.One2many("hr.job", "survey_id", string="Job Position")
+
+    @api.depends('survey_type')
+    @api.depends_context('uid')
+    def _compute_allowed_survey_types(self):
+        super()._compute_allowed_survey_types()
+        if self.env.user.has_group('hr_recruitment.group_hr_recruitment_interviewer') or \
+           self.env.user.has_group('survey.group_survey_user'):
+            self.allowed_survey_types = (self.allowed_survey_types or []) + [('recruitment', 'Recruitment')]
+
+    def get_formview_id(self, access_uid=None):
+        if self.survey_type == 'recruitment' and not (self.env['res.users'].browse(access_uid) or self.env.user).has_group('survey.group_survey_user'):
```

Comment:
In this case, I'd go for a few lines... Maybe it's just me, but as you'd still have to do something about its length, I'd go like this:

```suggestion
        if self.survey_type == 'recruitment':
            access_user = self.env['res.users'].browse(access_uid) if access_uid else self.env.user
            if not access_user.has_group('survey.group_survey_user'):
```